src/index.ts

```diff
@@ -166,6 +226,50 @@ async function main() {
     process.exit(1);
   }
 
+  try {
+    if (conf.useMercury && redis) {
+      const checkNetwork = "PUBLIC";
+      // initial token and userID not needed for integrity checks
+      const integrityCheckMercurySession = {
+        renewClientMaker,
+        backendClientMaker,
+        currentDataClientMaker,
+        backends,
+        credentials: {
+          TESTNET: {
+            email: conf.mercuryEmail,
+            password: conf.mercuryPassword,
+          },
+          // need to set this to the integrity check accounts for Mercury to remove entries periodically
+          PUBLIC: {
+            email: conf.mercuryIntegrityCheckEmail,
+            password: conf.mercuryIntegrityCheckPass,
+          },
+        },
+      };
+      const integrityCheckMercuryClient = new MercuryClient(
+        integrityCheckMercurySession,
+        logger,
+        register,
+        {
+          mercuryErrorCounter,
+          rpcErrorCounter,
+          criticalError,
+        },
+        redis
+      );
+      const stellarClient = new IntegrityChecker(
+        logger,
+        integrityCheckMercuryClient,
+        redis,
+        register
+      );
+      // await stellarClient.watchLedger(checkNetwork);
```

Comment:
Commented out right now because pubnet checks are causing issues in Mercury because of the deeper histories. This has prompted an architecture change on their end but in the mean time we can test using a testnet account. Leaving it commented here for now so that the preview deployment doesn't cause issues for them in the meantime.